language/Type-Classes.md

```diff
@@ -55,7 +55,29 @@ main = do
 
 ## Multi-Parameter Type Classes
 
-TODO. For now, see the section in [PureScript by Example](https://leanpub.com/purescript/read#leanpub-auto-multi-parameter-type-classes).
+A type class can have more than one parameter, and these type classes are called multi-parameter type classes. For example, we can define a class where a given input type A can always be injected into another type B, but projecting a specific type A out of B is not guaranteed:
+
+```purs
+class Inject a b where
+  inj :: a -> b
+  prj :: b -> Maybe a
+
+instance injectLeft :: Inject a (Either a b) where
+  inj = Left
+  prj = either Just (const Nothing)
+
+else instance injectRight :: Inject a b => Inject a (Either c b) where
+  inj = Right <<< inj
+  prj = either (const Nothing) prj
+
+else instance injectReflexive :: Inject a a where
```

Comment:
Shouldn't this (`injectReflexive`) be the first instance?
Suppose `a` is `Either Int String`, currently this would hit the `injectRight` case and then try `Inject (Either Int String) String` - which isn't quite what you might expect.

Reply:
Example taken from https://github.com/purescript/purescript-either/blob/v4.0.0/src/Data/Either/Inject.purs#L8-L10

Reply:
Maybe we can put this link below the example? This would increase the chance that a reader will notice this example differs from its source, should the source ever be changed/bug-fixed.

Would be nice to avoid making you come up with a whole-new example.

Reply:
Shall we resolve this in the `purescript-either` repo first? https://github.com/purescript/purescript-either/issues/43

Reply:
That issue has been resolved, it appears: https://github.com/purescript/purescript-either/issues/43